dadi/lib/storage/http.js

```diff
@@ -96,7 +103,17 @@ HTTPStorage.prototype.get = function () {
           break
       }
 
-      return reject(err)
+      if (err.statusCode === 404) {
+        return new Missing().get().then(stream => {
```

Comment:
I personally tend to not create `Promise` constructs when already inside one, which would allow you to get rid of the `catch`:

```js
return resolve(
  new Missing().get().then(stream => {
    this.notFound = true
    this.lastModified = new Date()

    return stream
  })
)
```

But this one is really just a matter of personal taste, so more than happy for it to stay as it is!